src/components/ProductVersionBuildConfigsEditPage/ProductVersionBuildConfigsEditPage.tsx

```diff
@@ -0,0 +1,264 @@
+    .map((buildConfigChange) => buildConfigChange.data);
+
+  const addBuildConfigChange = (buildConfig: BuildConfiguration, operation: IBuildConfigChange['operation']) =>
+    setBuildConfigChanges((buildConfigChanges) => {
+      const otherBuildConfigChanges = buildConfigChanges.filter(
+        (buildConfigChange) => buildConfigChange.data.id !== buildConfig.id
+      );
+      return [...otherBuildConfigChanges, { data: buildConfig, operation: operation }];
+    });
+
+  const cancelBuildConfigChange = (buildConfig: BuildConfiguration) =>
+    setBuildConfigChanges((buildConfigChanges) => {
+      return buildConfigChanges.filter((buildConfigChange) => buildConfigChange.data.id !== buildConfig.id);
+    });
+
+  const [isSubmitModalOpen, setIsSubmitModalOpen] = useState<boolean>(false);
+  const toggleSubmitModal = () => setIsSubmitModalOpen((isSubmitModalOpen) => !isSubmitModalOpen);
+
+  const [isCancelAllModalOpen, setIsCancelAllModalOpen] = useState<boolean>(false);
+  const toggleCancelAllModal = () => setIsCancelAllModalOpen((isCancelAllModalOpen) => !isCancelAllModalOpen);
+
+  useEffect(() => {
+    serviceContainerProductVersionRunner({ serviceData: { id: productVersionId } });
+  }, [serviceContainerProductVersionRunner, productVersionId]);
+
+  useQueryParamsEffect(
+    ({ requestConfig } = {}) =>
+      serviceContainerProductVersionBuildConfigsRunner({ serviceData: { id: productVersionId }, requestConfig }),
+    { componentId: componentIdProductVersionBuildConfigs }
+  );
+
+  useQueryParamsEffect(serviceContainerProjectBuildConfigsRunner, { componentId: componentIdProjectBuildConfigs });
+
+  useTitle(
+    generatePageTitle({
+      pageType: 'Edit',
+      serviceContainer: serviceContainerProductVersion,
+      firstLevelEntity: 'Product',
+      nestedEntity: 'Version',
+      entityName: `Build Configs of ${serviceContainerProductVersion.data?.version} ${PageTitles.delimiterSymbol} ${serviceContainerProductVersion.data?.product.name}`,
+    })
+  );
+
+  return (
+    <PageLayout
+      title={`Add and remove Build Configs${productVersion ? ' in ' + productVersion : ''}`}
+      actions={[
+        <Button
+          key="submit-btn"
+          variant="primary"
+          onClick={() => {
+            toggleSubmitModal();
+          }}
+          isDisabled={!buildConfigChanges.length}
+        >
+          Submit changes
+        </Button>,
+      ]}
+      description={
+        <>
+          <div>
+            Update the list of Build Configs associated with the Version by removing existing ones or adding new ones. All the
+            Build Configs marked to be removed or added are listed in the Changes Summary section. To submit the changes, use the
+            Submit button.
+          </div>
+          <div className={css(!buildConfigChanges.length && 'visibility-hidden')}>
+            <ExclamationTriangleIcon color="#F0AB00" /> Submit the changes to apply them.
```

Comment:
-----
![image](https://github.com/project-ncl/pnc-web-ui-react/assets/17566891/9d550e00-b8f6-477c-b437-e1de9fbc75e2)
-----

I miss some top margin here, but I understand it will make "whitespace placeholder" bigger, nice solution could be to put it directly next to the button:

-----
![image](https://github.com/project-ncl/pnc-web-ui-react/assets/17566891/78fdce99-fbb3-4145-841e-103e7ab4d2ed)
-----

Reply:
done